app-service-msi-keyvault-python-v3/README.md

```diff
@@ -0,0 +1,189 @@
+   - `/` : The MSI sample itself
+
+## Installation on Azure
+
+1. Set the `KEY_VAULT_URL` environment variable using the "Application Settings" of your WebApp.
+
+1. Connect to the [Kudu console](https://github.com/projectkudu/kudu/wiki/Kudu-console) and install the dependencies. If you installed the Python 3.6.2x86 extension, the command line will be:
+
+```shell
+D:\home\python362x86\python.exe -m pip install -r D:\home\site\wwwroot\requirements.txt
+```
+
+> For automation purpose, you might use the [Kudu RestAPI](https://github.com/projectkudu/kudu/wiki/REST-API#command)
+
+3. This repo is ready to be deployed using local git. Read this tutorial to get more information on [how to push using local git with CLI 2.0](https://docs.microsoft.com/azure/app-service/app-service-web-get-started-python#push-to-azure-from-git)
+
+## At a glance
+
+Using the `MSIAuthentication` class will autodetect if you're on a WebApp and get the token from the MSI endpoint. Then, the code is simply:
+
+```python    
+    credentials = MSIAuthentication(
+        resource='https://vault.azure.net'
+    )
+    key_vault_client = KeyVaultClient(
+        credentials
+    )
+
+    key_vault_uri = os.environ.get("KEY_VAULT_URI")
+
+    secret = key_vault_client.get_secret(
+        key_vault_uri,  # Your KeyVault URL
+        "secret",       # Name of your secret. If you followed the README 'secret' should exists
+        ""              # The version of the secret. Empty string for latest
+    )
+```
+
+If you want to execute this same code in your local environment machine, just replace the `MSIAuthentication` class with `ServicePrincipalCredentials` or any credentials class available for Python. 
+An overview of credentials features is available here: [https://docs.microsoft.com/python/azure/python-sdk-azure-authenticate](https://docs.microsoft.com/python/azure/python-sdk-azure-authenticate)
+
+If you need a fallback mechanism to allow this code to switch automatically from MSI to another approach, you can test for environment variables:
+
+```python
+if "APPSETTING_WEBSITE_SITE_NAME" in os.environ:
+    return MSIAuthentication(
+        resource='https://vault.azure.net'
+    )
+else:
+    return ServicePrincipalCredentials(
+        client_id=os.environ['AZURE_CLIENT_ID'],
+        secret=os.environ['AZURE_CLIENT_SECRET'],
+        tenant=os.environ['AZURE_TENANT_ID'],
+        resource='https://vault.azure.net'
+    )
+```
+
+## Summary
+
+The web app was successfully able to get a secret at runtime from Azure Key Vault using your developer account during development, and using MSI when deployed to Azure, without any code change between local development environment and Azure. 
+As a result, you did not have to explicitly handle a service principal credential to authenticate to Azure AD to get a token to call Key Vault. You do not have to worry about renewing the service principal credential either, since MSI takes care of that.
+
+## Azure Functions
```

Comment:
I'm not sure we want to include info about Azure Functions in this sample.  It might just confuse the user.

Reply:
removed info about Azure Functions